FFC testzone

IceFreez3r opened the pull request.

IceFreez3r committed:
Step 2 Performance 1/2

Removed scores-vector. Working on the trace_score (double) should be sufficient here.
Stored map[s +- i] in next_spectrum to prevent multiple accesses. Was probably already optimized by the compiler.

Not tested!

IceFreez3r committed:
const readded

IceFreez3r committed:
Ranged based for-Loops, OMP-Checks removed

Changed all for-Loops to ranged based for-Loops where CLion recommended it
Removed checks for OMP: #pragma omp parallel does nothing without OMP

Changed a few occurences of push_back to emplace_back where possible

IceFreez3r committed:
Small change

Small changes similar to the commits before

IceFreez3r committed:
less branches in checkFreatureQuality_

Instead of changing feature_ok to false and returning it at the end, just instant return false. Prevents multiple if(feature_ok) branches.

IceFreez3r committed:
Forgot one return

IceFreez3r committed:
Update FeatureFinderAlgorithmPicked.cpp

cbielow commented on src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmPicked.cpp:
```diff
@@ -401,9 +394,9 @@ namespace OpenMS
         d.trimRight(intensity_percentage_optional_);
 
-        for (IsotopeDistribution::Iterator it = d.begin(); it != d.end(); ++it)
+        for (auto & it : d)
```
dont name it `it` since its not an iterator. Use a name corresponding to the type.

IceFreez3r replied:
Renamed to peak

cbielow commented on src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmPicked.cpp:
```diff
@@ -431,17 +424,17 @@ namespace OpenMS
         }
         isotope_distributions_[index].max = max;
-        for (Size i = 0; i < isotope_distributions_[index].intensity.size(); ++i)
+        for (double & i : isotope_distributions_[index].intensity)
```
naming convention `double& i`

cbielow commented on src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmPicked.cpp:
```diff
@@ -573,13 +566,13 @@ namespace OpenMS
         FeatureMap seed_map;
         seed_map.reserve(seeds.size());
-        for (Size i = 0; i < seeds.size(); ++i)
+        for (auto & seed : seeds)
```
`&` to the type...
happens in other places as well.

cbielow commented on src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmPicked.cpp:
```diff
@@ -854,28 +840,27 @@ namespace OpenMS
+        Size seed_nr = iter.first;
         bool is_used = false;
-        for (Size i = 0; i < seeds_contained.size(); ++i)
+        for (unsigned long i : seeds_contained)
```
`for (Size i : ...`

IceFreez3r replied:
Done

cbielow commented on src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmPicked.cpp:
```diff
@@ -854,28 +840,27 @@ namespace OpenMS
       // features_ list.
       std::vector<Size> seeds_contained;
-      for (std::map<Size, Feature>::iterator iter = tmp_feature_map.begin(); iter != tmp_feature_map.end(); ++iter)
+      for (auto & iter : tmp_feature_map)
```
`iter` is not an iterator?
use `f` if its a feature

IceFreez3r replied:
Done

cbielow commented on src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmPicked.cpp:
```diff
@@ -854,28 +840,27 @@ namespace OpenMS
 
           std::vector<Size> curr_seed = seeds_in_features[seed_nr];
-          for (Size k = 0; k < curr_seed.size(); ++k)
+          for (unsigned long k : curr_seed)
```
Size k

IceFreez3r replied:
Done

IceFreez3r committed:
Coding conventions

IceFreez3r committed:
Uncommented continue()-s

There werde multiple occurences of commented continue()-s. Now there are uncommented and the following else parts aren't necessary anymore and were removed
